src/script/lua_api/l_http.cpp

```diff
@@ -44,10 +44,13 @@ void ModApiHttp::read_http_fetch_request(lua_State *L, HTTPFetchRequest &req)
 	getstringfield(L, 1, "url", req.url);
 	lua_getfield(L, 1, "user_agent");
 	if (lua_isstring(L, -1))
-		req.useragent = getstringfield_default(L, 1, "user_agent", "");
+		req.useragent = lua_tostring(L, -1);
 	lua_pop(L, 1);
 	req.multipart = getboolfield_default(L, 1, "multipart", false);
-	req.timeout = getintfield_default(L, 1, "timeout", 3) * 1000;
+	lua_getfield(L, 1, "timeout");
+	if (lua_isnumber(L, -1))
+		req.timeout = lua_tointeger(L, -1) * 1000;
+	lua_pop(L, 1);
```

Comment:
Alternatively, shorter:
```C++
if (getintfield(L, 1, "timeout", req.timeout))
	req.timeout *= 1000;
```
`nil` values are ignored (i.e. old value is kept), and a proper error message is shown when the API is used wrongly (i.e. wrong types).
Same would be possible for `req.useragent`.